### PR TITLE
feat: allowUnsafeRegex in options

### PR DESCRIPTION
**Pull request by BCsabaEngine**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by BCsabaEngine** [comment]
bump find-my-way v5.3.0 because two fix arrived since 5.1.0

https://github.com/delvedor/find-my-way/commit/a7c4216449b048ca18cbec909e0ea5f1190fd73c

https://github.com/delvedor/find-my-way/commit/efbc8a6a202420aed9eb36c40e22003b4f15959e

File: package.json [hunk 1]
```diff
@@ -180,7 +180,7 @@
     "abstract-logging": "^2.0.1",
     "avvio": "^8.1.0",
     "fastify-error": "^1.0.0",
-    "find-my-way": "^5.1.0",
+    "find-my-way": "^5.3.0",
```
Comment (jsumners) [inline package.json:183]: This is unnecessary.

**Review by mcollina — APPROVED** [review]
lgtm

File: package.json (hunk 1 above)
Comment (BCsabaEngine) [reply package.json:183]: Should I revoke or you can skip when merge?

Comment (mcollina) [reply package.json:183]: No need to change this at this point

**Review by jsumners — APPROVED** [review]

**Pull request merged**